utility/src/main/java/org/oppia/android/util/logging/firebase/FirebaseEventLogger.kt

```diff
@@ -30,8 +31,17 @@ class FirebaseEventLogger(
     firebaseAnalytics.logEvent(eventLog.actionName.toString(), bundle)
     // TODO(#3792): Remove this usage of Locale.
     firebaseAnalytics.setUserProperty(COUNTRY_USER_PROPERTY, Locale.getDefault().displayCountry)
-    firebaseAnalytics.setUserProperty(
-      NETWORK_USER_PROPERTY, connectivityManager.activeNetworkInfo.typeName
-    )
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+      val networkCapabilities = connectivityManager.getNetworkCapabilities(
+        connectivityManager.activeNetwork
+      )
+      firebaseAnalytics.setUserProperty(
+        NETWORK_USER_PROPERTY, networkCapabilities?.transportInfo.toString()
```

Comment:
@BenHenning Do you have any preferences on what should we add to this property?
As `NetworkCapabilities` doesn't provide many readable items. Things that we can use are `transportTypes` or `capabilities`, but the `NetworkCapabilities` API is designed in such a way that these are not available to use.

Reply:
I'm not sure--what's it for? I'm having difficulty understanding the question since I'm not sure that is supposed to be passed here for Firebase.

@Sarthak2601 might have some thoughts, though I think you should cross-reference with Firebase documentation to determine how we're supposed to correctly initialize this on Android.

Reply:
Updated to logging the network connection status.